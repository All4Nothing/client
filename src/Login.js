import React, { useEffect, useState } from 'react';
import axios from 'axios';

function Login() {
    const [inputId, setInputId] = useState('');
    const [inputPw, setInputPw] = useState('');

    const handleInputId = (event) => { //event: React.ChangeEvent
        setInputId(event.target.value);
    }

    const handleInputPw = (event) => {
        setInputPw(event.target.value);
    }

    
    const handleKeyPress = (event) => {
        if (event.key === "Enter") {
            onClickLogin();
        }
    }

    const onClickLogin = async () => {
        try {
            console.log('click login')
            console.log('ID : ', inputId)
            console.log('PW : ', inputPw)
            const response = await axios.post('http://127.0.0.1:8000/api/auth/login/', {
                'username': inputId,
                'password': inputPw
            })
            if (response.status === 200) {
                console.log(response)
                document.location.href = '/'
            }
        } catch (error) {
            console.dir(error)
        }
    }


    return(
        <div>
            <h1>Login</h1>
            <div>
                <label htmlFor="input_id">ID:</label>
                <input type = "text" id = "input_id" value = {inputId} onChange = {handleInputId} />
                
            </div>
            <div>
                <label htmlFor="input_pw">PassWord:</label>
                <input type = "password" id = "input_pw" value = {inputPw} onChange = {handleInputPw} onKeyPress = {handleKeyPress} />    
                
            </div>
            <div>
                <button type="button" onClick={onClickLogin}>Login</button>
            </div>
        </div>
    )
}

export default Login;