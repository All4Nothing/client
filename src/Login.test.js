import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import Login from "./Login"

describe('<Login />', () => {
    it('matches snapshot', () => {
        const utils = render(<Login />);
        expect(utils.container).toMatchSnapshot();
    });
    it('has two input', () => {
        render(<Login />);
        screen.getByLabelText('ID:');
        screen.getByLabelText('PassWord:')
    });
    it('text change', () => {
        render(<Login />);
        const IdBtn = screen.getByLabelText('ID:');
        const PwBtn = screen.getByLabelText('PassWord:');
        fireEvent.change(IdBtn, { target: { value: '김용주' } });
        fireEvent.change(PwBtn, { target: { value: 'kyj130656' } });
        expect(IdBtn.value).toBe('김용주');
        expect(PwBtn.value).toBe('kyj130656');
    });
})