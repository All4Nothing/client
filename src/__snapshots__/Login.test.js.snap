// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches snapshot 1`] = `
<div>
  <div>
    <h1>
      Login
    </h1>
    <div>
      <label
        for="input_id"
      >
        ID:
      </label>
      <input
        id="input_id"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="input_pw"
      >
        PassWord:
      </label>
      <input
        id="input_pw"
        type="password"
        value=""
      />
    </div>
    <div>
      <button
        type="button"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
